File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int profit = 0;
        int purchase = prices[0];
        for(int i=0; i<prices.size(); i++){
            if(prices[i] < purchase) purchase = prices[i];
            profit = max(profit, prices[i] - purchase);
        }
        return profit;
    }
};

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int maxi = 0;
        int bb = prices[0];

        for(int i=0 ; i<prices.length;i++){
            if(prices[i]>bb){
                maxi= Math.max(maxi,prices[i]-bb);
            }
            bb = Math.min(bb,prices[i]);
        }
        return maxi;
    }
}